src/net_processing.cpp

```diff
@@ -4039,6 +4076,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
             // Erase this entry from tx_process_time (it may be added back for
             // processing at a later time, see below)
             tx_process_time.erase(tx_process_time.begin());
+            state.m_tx_download.m_tx_announced[txid] = std::chrono::microseconds::zero();
```

Comment:
I think this line handles a tricky corner case. Can we add comment to reduce the cognitive burned on code reviewer?
What I think happens is (correct me if I'm wrong):
- request from X1, not found
- request from X2, ignores us for 1 minute
- request from X3, notfound
- this line prevents us from querying X2 again (we would if this line is deleted)

Now that I wrote this, one could simply check the presence of m_tx_in_flight for that peer, and if it is in flight, not consider for re-query?

Reply:
I think the idea was to stop you from doing `INV x; NOTFOUND x; INV x` repeatedly and making it harder for you to query other peers. But I think this introduces a bug -- the `zero` entry never gets cleared from `m_tx_announced`, because that's only cleared based on what's in `in_flight` and `process_time` and the txid isn't in either of those anymore.